src/atomate2/vasp/sets/core.py

```diff
@@ -800,6 +808,129 @@ def get_kpoints_updates(
         return {"reciprocal_density": self.reciprocal_density}
 
 
+@dataclass
+class MDSetGenerator(VaspInputGenerator):
+    """
+    Class to generate VASP molecular dynamics input sets.
+
+    Parameters
+    ----------
+    ensemble
+        Molecular dynamics ensemble to run. Options include `nvt`, `nve`, `npt`, and
+        `nph`.
+    start_temp
+        Starting temperature. The VASP `TEBEG` parameter.
+    end_temp
+        Final temperature. The VASP `TEEND` parameter.
+    nsteps
+        Number of time steps for simulations. The VASP `NSW` parameter.
+    time_step
+        The time step (in femtosecond) for the simulation. The VASP `POTIM` parameter.
+    spin_polarized
+        Whether to do spin polarized calculations. The VASP `ISPIN` parameter.
+    """
+
+    ensemble: str = "nvt"
+    start_temp: float = 300
+    end_temp: float = 300
+    nsteps: int = 1000
+    time_step: int = 2
+    spin_polarized: bool = False
+
+    def get_incar_updates(
+        self,
+        structure: Structure,
+        prev_incar: dict = None,
+        bandgap: float = 0,
+        vasprun: Vasprun = None,
+        outcar: Outcar = None,
+    ) -> dict:
+        """
+        Get updates to the INCAR for a molecular dynamics job.
+
+        Parameters
+        ----------
+        structure
+            A structure.
+        prev_incar
+            An incar from a previous calculation.
+        bandgap
+            The band gap.
+        vasprun
+            A vasprun from a previous calculation.
+        outcar
+            An outcar from a previous calculation.
+
+        Returns
+        -------
+        dict
+            A dictionary of updates to apply.
+        """
+        updates = self._get_ensemble_defaults(self.ensemble)
+
+        # Based on pymatgen.io.vasp.sets.MPMDSet. Changes include:
+        # LREAL: True -> Auto
+        # ADDGRID: True -> False
+        # EDIFF_PER_ATOM: 0.00001, removed in favor of EDIFF
+        updates.update(
+            {
+                "ENCUT": None,  # None to use VASP default
+                "TEBEG": self.start_temp,
+                "TEEND": self.end_temp,
+                "NSW": self.nsteps,
+                "POTIM": self.time_step,
+                "ISPIN": 2 if self.spin_polarized else 1,
+                "LSCALU": False,
+                "LCHARG": False,
+                "LPLANE": False,
+                "LWAVE": True,
+                "ISMEAR": 0,
+                "NELMIN": 4,
+                "LREAL": "Auto",
+                "BMIX": 1,
+                "MAXMIX": 20,
+                "NELM": 500,
+                "NSIM": 4,
+                "ISYM": 0,
+                "IBRION": 0,
+                "NBLOCK": 1,
+                "KBLOCK": 100,
+                "PREC": "Normal",
+                "LDAU": False,
```

Comment:
Why is DFT+U disabled? Shouldn't we just use the default U values from pymatgen? In that case just remove this tag and they will be applied automatically!

Reply:
From a code hygiene perspective, @munrojm and I were discussing recently that there's a big advantage to having all calculations map cleanly to a pymatgen input set, since this greatly simplifies the task validation logic. If it's at all possible to inherit the settings from a specific input set, rather than copying manually, I think this would be very useful.

Reply:
@mkhorton Just to make sure I understand, by `simplifies the task validation logic`, are you referring to the emmet building process or something else? And can you expand a bit on this...

Reply:
Also, @mkhorton could you clarify what you mean by "inherit the settings from a specific input set, rather than copying manually"?